src/app/editor/_components/createPostForm/CreatePostForm.tsx

```diff
@@ -0,0 +1,313 @@
+                    sx={{
+                      mb: 2,
+                      "& .MuiInputBase-inputMultiline": { minHeight: 64 },
+                    }}
+                  />
+                  <Box
+                    sx={{
+                      mb: 2,
+                    }}
+                  >
+                    {value.map((actor, index) => (
+                      <Chip
+                        key={index}
+                        label={actor}
+                        onDelete={() => handleRemoveActors(index)}
+                        sx={{
+                          backgroundColor: "#f0f0f0",
+                          fontWeight: "bold",
+                          width: "fit-content",
+                          padding: "4px 8px",
+                          margin: "4px",
+                        }}
+                      />
+                    ))}
+                  </Box>
+                </FormControl>
+              );
+            }}
+          />
+        )}
+
+        <Controller
+          control={control}
+          name="tags"
+          defaultValue={createPostFormData.tags}
+          render={({ field: { value, onChange } }) => (
+            <FormControl fullWidth>
+              <Typography variant="h6" gutterBottom>
+                Select relevant tags
+              </Typography>
+              <Grid container spacing={1}>
+                {INTEREST_ARRAY.map((interest) => {
+                  const isSelected = value?.includes(interest);
+                  return (
+                    <Grid key={interest}>
+                      <Chip
+                        label={interest}
+                        onClick={() => {
+                          if (isSelected) {
+                            onChange(
+                              value?.filter((item) => item !== interest),
+                            );
+                          } else {
+                            onChange([...(value ?? []), interest]);
+                          }
+                        }}
+                        color={isSelected ? "primary" : "default"}
+                        variant={isSelected ? "filled" : "outlined"}
+                      />
+                    </Grid>
+                  );
+                })}
+              </Grid>
+              {errors?.tags?.message && (
+                <Typography color="error" variant="body2">
+                  {errors.tags.message}
+                </Typography>
+              )}
+            </FormControl>
+          )}
+        />
+      </DialogContent>
+      <DialogActions sx={{ justifyContent: "space-between", padding: 3 }}>
+        <Button
+          variant="outlined"
+          onClick={handleClose}
+          sx={{ width: "100px" }}
+        >
+          Cancel
+        </Button>
+        <Button
+          variant="contained"
+          onClick={handleSubmit((data) => console.log(data))}
+          sx={{ width: "100px" }}
+        >
+          Save
```

Comment:
Lets call it Create instead of Save

Reply:
Also on click of of Create we can redirect to /editor without saving anything to backend. And on editor page on click of back we redirect to previous page

Reply:
but how do we pass the metadata with the route, that is why i was creating a entry in the backend when the user finalise the metadata cause a empty post is still a post and when we redirect to that editor we fetch from the backend for that specific id

Reply:
We can use prop or query params for that purpose. In that way we will make only one api call when the post is published. A post is only a post when it is published. It can be a draft before that but not a post

Reply:
i meant not a post , but a seperate collection will made for that only published ones goes in the post model

Reply:
not sure i understand. We will need to connect on this